contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+            IVault(vaultAddress).rebase();
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and swap them for a
+     *      supported stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from.
+     */
+    function _harvest(address _strategyAddr) internal {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        address[] memory rewardTokenAddresses = strategy
+            .getRewardTokenAddresses();
+
+        if (rewardTokenAddresses.length > 0) {
+            strategy.collectRewardTokens();
+        }
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function _swap() internal {
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            _swap(swapTokens[i]);
+        }
+    }
+
+    /**
+     * @dev Swap a record token for stablecoins on Uniswap. The token must have
+     *       a registered price feed with the price provider.
+     * @param _swapToken Address of the token to swap.
+     */
+    function _swap(address _swapToken) internal {
+        RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+        require(
+            tokenConfig.uniswapV2CompatibleAddr != address(0),
+            "Swap token is missing token configuration."
+        );
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+
+        IERC20 swapToken = IERC20(_swapToken);
+        uint256 balance = swapToken.balanceOf(address(this));
+
+        if (balance == 0) {
+            return;
+        }
+
+        uint256 maxBalanceToSwap = balance;
+        if (tokenConfig.liquidationLimit != 0) {
+            maxBalanceToSwap = Math.min(balance, tokenConfig.liquidationLimit);
+        }
+
+        // This'll revert if there is no price feed
+        uint256 oraclePrice = IOracle(priceProvider).price(_swapToken);
+        // Oracle price is 1e8, USDT output is 1e6
+        uint256 minExpected = (maxBalanceToSwap *
+            oraclePrice *
+            (1e4 - tokenConfig.allowedSlippageBps)).scaleBy( // max allowed slippage
+            6,
+            Helpers.getDecimals(_swapToken) + 8
+        ) / 1e4; // fix the max slippage decimal position
+
+        // Uniswap redemption path
+        address[] memory path = new address[](3);
+        path[0] = _swapToken;
+        path[1] = IUniswapV2Router(tokenConfig.uniswapV2CompatibleAddr).WETH();
+        path[2] = usdtAddress;
+
+        IUniswapV2Router(tokenConfig.uniswapV2CompatibleAddr)
+            .swapExactTokensForTokens(
+                maxBalanceToSwap,
+                minExpected,
+                path,
+                address(this),
+                block.timestamp
+            );
+
+        IERC20 usdt = IERC20(usdtAddress);
+        uint256 usdtBalance = usdt.balanceOf(address(this));
+
+        // When governor calls the swap function send full reward amount to the Vault
+        if (isGovernor()) {
```

Comment:
@DanielVF addressing your comment that was lost by the code refactor: 
```
If the above calculations return USDT amounts for vault and harvester, then this can become a simple check to see if there anything to send to the harvester, and aways send to the vault.

The actual isGoverner logic can live up in the math/logic above, rather than here in the doing.
```

I've refactored this part a bit, since the share of the proceeds between vault & harvester didn't really belong outside of the if clause scope. Does this version seem ok to you or would you still rather see it refactored without the if statement?

I like how it is right now since it seems readable to me.

Reply:
Yes, this works structurally.